src/Dialect/Krnl/KrnlOps.cpp

```diff
@@ -511,11 +511,15 @@ void KrnlInstrumentOp::build(mlir::OpBuilder &builder, OperationState &state,
   // Put only the opName part in the opID within its size
   strncpy((char *)&opID, opName + 5, sizeof(decltype(opID)) - 1);
   IntegerAttr attr = builder.getI64IntegerAttr(opID);
-  auto tagAttr = builder.getI64IntegerAttr(tag);
-  StringAttr nameAttr = builder.getStringAttr(StringRef(opName));
-  state.addAttribute("opName", nameAttr);
+  StringAttr nodeNameAttr =
+      op->getAttrOfType<::mlir::StringAttr>("onnx_node_name");
+  IntegerAttr tagAttr = builder.getI64IntegerAttr(tag);
+  StringAttr opNameAttr = builder.getStringAttr(StringRef(opName));
+  state.addAttribute("opName", opNameAttr);
   state.addAttribute("opID", attr);
   state.addAttribute("tag", tagAttr);
+  if (nodeNameAttr)
+    state.addAttribute("nodeName", nodeNameAttr);
```

Comment:
Can we use default builder instead of addAttribute here?

Reply:
@chentong319 I updated to  use the default builder. Thanks!